### PR TITLE
Integrate Theme Toggling

### PR DESCRIPTION
**Pull request by 12944qwerty**
This literally took 2 minutes to do lol

If it needs anything else, just let me know I guess. I also kept the beta code in there in case anyone wanted it to stay there anyway :P

pretty self-explanatory what this does:
 - toggles themes in the built-in themes settings

### CODE REVIEW

**Comment by 12944qwerty** [comment]
BTW, don't merge yet.

I still have to add commands and fix uninstall buttons

File: src/Powercord/plugins/pc-moduleManager/components/manage/Themes.jsx [hunk 1]
```diff
@@ -56,44 +57,64 @@ class Themes extends Base {
...
-        <Button className='developerPortalCta-3QrlHn' onClick={() => this.setState({ tryBeta: true })}>
-          Try the beta
-        </Button>
-      </div>
+      <InstalledProduct
+        product={item.manifest}
+        isEnabled={powercord.styleManager.isEnabled(item.entityID)}
+        onToggle={async v => {
+          await this._toggle(item.entityID, v);
+          this.forceUpdate();
```
Comment (LandenStephenss) [inline src/Powercord/plugins/pc-moduleManager/components/manage/Themes.jsx:61]: just make this a functional component where you pass some functions and use useState??

**Comment by 12944qwerty** [comment]
I've been trying for a bit and I'm unable to figure out how to fix the uninstall button.
I've added a bunch of errorhandlers and all, but I can't seem to figure out where the crash is happening.

I'll continue debugging, but just pull this request for now. I'll open a new one if I don't figure it out before this is pulled.

File: src/Powercord/plugins/pc-moduleManager/components/manage/Themes.jsx [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
+
 const ThemeSettings = require('./ThemeSettings');
 const QuickCSS = require('./QuickCSS');
 const Base = require('./Base');
+const InstalledProduct = require('../parts/InstalledProduct');
 
-const BETA_IDENTIFIER = 3542355018683011159509n;
-const FEEDBACK_IDENTIFIER = 3758304876382993109949n;
+// const BETA_IDENTIFIER = 3542355018683011159509n;
+// const FEEDBACK_IDENTIFIER = 3758304876382993109949n;
```
Comment (asportnoy) [inline src/Powercord/plugins/pc-moduleManager/components/manage/Themes.jsx]: Why leave these commented? Just remove them if you don't need them.

File: src/Powercord/plugins/pc-moduleManager/components/manage/Themes.jsx [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
-function encodeIdentifier (i) {
-  const b = Buffer.alloc(8);
-  b.writeBigInt64BE((i - 69n) / 420n);
-  return b.toString('base64').replace(/=/g, '');
-}
+// function encodeIdentifier (i) {
+//   const b = Buffer.alloc(8);
+//   b.writeBigInt64BE((i - 69n) / 420n);
+//   return b.toString('base64').replace(/=/g, '');
+// }
```
Comment (asportnoy) [inline src/Powercord/plugins/pc-moduleManager/components/manage/Themes.jsx]: Same as above

File: src/Powercord/plugins/pc-moduleManager/components/manage/Themes.jsx [hunk 4]
```diff
@@ -56,44 +60,64 @@ class Themes extends Base {
...
+  //     <Button className='developerPortalCta-3QrlHn' onClick={() => this.setState({ tryBeta: true })}>
+  //       Try the beta
+  //     </Button>
+  //   </div>
+  // );
+
+  //   return this.getItems().map((theme) => {
+  //     return this.renderItem(theme);
+  //   });
+  // }
```
Comment (asportnoy) [inline src/Powercord/plugins/pc-moduleManager/components/manage/Themes.jsx]: Same as above

**Review by asportnoy — CHANGES_REQUESTED** [review]

**Review by asportnoy — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -27,7 +27,7 @@
     "unzip-crx": "^0.2.0"
   },
   "devDependencies": {
-    "eslint": "^7.8.1",
```
Comment (asportnoy) [inline package.json:30]: What was the reason for this update?

File: src/Powercord/plugins/pc-moduleManager/commands/enable.js [hunk 6]
```diff
@@ -1,22 +1,29 @@
 module.exports = {
   command: 'enable',
-  description: 'Allows you to re-/enable a selected plugin from the given list.',
-  usage: '{c} [ plugin ID ]',
+  description: 'Allows you to re-/enable a selected plugin/theme from the given list.',
```
Comment (asportnoy) [inline src/Powercord/plugins/pc-moduleManager/commands/enable.js]:
```suggestion
  description: 'Enable a plugin/theme',
```

File: src/Powercord/plugins/pc-moduleManager/commands/disable.js [hunk 7]
```diff
@@ -1,25 +1,34 @@
+const { CORE_PLUGINS } = require('powercord/constants');
+
 module.exports = {
   command: 'disable',
-  description: 'Allows you to disable a selected plugin from the given list.',
-  usage: '{c} [ plugin ID ]',
+  description: 'Allows you to disable a selected plugin/theme from the given list.',
```
Comment (asportnoy) [inline src/Powercord/plugins/pc-moduleManager/commands/disable.js]:
```suggestion
  description: 'Disable a plugin/theme',
```

File: src/Powercord/plugins/pc-moduleManager/commands/disable.js [hunk 8]
```diff
@@ -30,21 +39,31 @@ module.exports = {
...
         .map(plugin => ({
           command: plugin.entityID,
           description: plugin.manifest.description
-        })),
+        })), ...themes
+        .filter(theme => theme.entityID.toLowerCase().includes(args[0] && args[0].toLowerCase()) && powercord.styleManager.isEnabled(theme.entityID))
+        .map(theme => ({
+          command: theme.entityID,
+          description: `Theme - ${theme.manifest.description}`
+        })) ],
```
Comment (asportnoy) [inline src/Powercord/plugins/pc-moduleManager/commands/disable.js]: You should try to reduce the amount of duplicate code in this function.

File: src/Powercord/plugins/pc-moduleManager/commands/enable.js [hunk 9]
```diff
@@ -29,17 +36,26 @@ module.exports = {
...
           command: plugin.entityID,
-          description: plugin.manifest.description
-        })),
+          description: `Plugin - ${plugin.manifest.description}`
+        })), ...themes
+        .filter(theme => theme.entityID.toLowerCase().includes(args[0] && args[0].toLowerCase()) && !powercord.styleManager.isEnabled(theme.entityID))
+        .map(theme => ({
+          command: theme.entityID,
+          description: `Theme - ${theme.manifest.description}`
+        })) ],
```
Comment (asportnoy) [inline src/Powercord/plugins/pc-moduleManager/commands/enable.js]: You should try to reduce the amount of duplicate code in this function.

**Review by asportnoy — CHANGES_REQUESTED** [review]

**Comment by aetheryx** [comment]
reopened @ https://github.com/powercord-org/powercord/pull/670

**Pull request closed without merging**